packages/ai-core/src/browser/ai-core-preferences.ts

```diff
@@ -63,26 +63,56 @@ export const aiCorePreferenceSchema: PreferenceSchema = {
+              - `modelId` (optional): The model ID to match\n\
+              - `providerId` (optional): The provider ID to match (e.g., huggingface, openai, ollama, llamafile)\n\
+              - `agentId` (optional): The agent ID to match\n\
+            - `requestSettings`: Model-specific settings as key-value pairs\n\
+            - `clientSettings`: Client-side message handling settings:\n\
+              - `keepToolCalls` (boolean): Whether to keep tool calls in the context\n\
+              - `keepThinking` (boolean): Whether to keep thinking messages\n\
+            Settings are matched based on specificity (agent: 100, model: 10, provider: 1 points).\n\
             Refer to [our documentation](https://theia-ide.org/docs/user_ai/#custom-request-settings) for more information.'),
             type: 'array',
             items: {
                 type: 'object',
                 properties: {
-                    modelId: {
-                        type: 'string',
-                        description: 'The model id'
+                    scope: {
+                        type: 'object',
+                        properties: {
+                            modelId: {
+                                type: 'string',
+                                description: 'The (optional) model id'
+                            },
+                            providerId: {
+                                type: 'string',
+                                description: 'The (optional) provider id to apply the settings to. If not set, the settings will be applied to all providers.',
+                            }, agentId: {
+                                type: 'string',
+                                description: 'The (optional) agent id to apply the settings to.',
+                            },
+                        }
                     },
                     requestSettings: {
                         type: 'object',
                         additionalProperties: true,
                         description: 'Settings for the specific model ID.',
                     },
-                    providerId: {
-                        type: 'string',
-                        description: 'The (optional) provider id to apply the settings to. If not set, the settings will be applied to all providers.',
+                    clientSettings: {
+                        type: 'object',
+                        additionalProperties: false,
+                        description: 'Client settings for how to handle messages that are send back to the llm.',
+                        properties: {
+                            keepToolCalls: {
+                                type: 'boolean',
+                                default: false
+                            },
+                            keepThinking: {
+                                type: 'boolean',
+                                default: false
+                            }
```

Comment:
Can you describe what each of these settings do exactly? As a user I don't have any idea why and when I would want to change them to `true`